### PR TITLE
Refactor mock_cgroup_commands

### PR DESCRIPTION
**Pull request by narrieta**
Split mock_cgroup_commands into MockEnvironment (which is generic) and mock_cgroup_environment (which mocks commands and files specific to cgroups)

### CODE REVIEW

File: tests/common/mock_cgroup_commands.py [hunk 1]
```diff
@@ -1,206 +0,0 @@
-# -*- coding: utf-8 -*-
```
Comment (narrieta) [inline tests/common/mock_cgroup_commands.py:1]: This file was renamed to mock_cgroup_environment.py. In the _MOCKED* arrays stay the same in the new file and the mocking logic is now moved to mock_environment.py

File: tests/common/mock_environment.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+import os
+import re
+import shutil
+import subprocess
+
+from azurelinuxagent.common.utils import fileutil
+from tests.tools import patch, patch_builtin
+
+
+class MockEnvironment:
```
Comment (narrieta) [inline tests/common/mock_environment.py:27]: As new functionality was added, the mocking logic was getting too long to be defined in a single function so I decided to make this a proper class instead.

File: tests/common/mock_environment.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        self._original_open = open
+
+        self.patchers = [
+            patch_builtin("open", side_effect=self._mock_open),
+            patch("subprocess.Popen", side_effect=self._mock_popen),
+            patch("os.path.exists", side_effect=self._mock_path_exists),
+            patch("azurelinuxagent.common.utils.fileutil.mkdir", side_effect=self._mock_mkdir)
+        ]
+
+    def __enter__(self):
```
Comment (narrieta) [inline tests/common/mock_environment.py:76]: Previously the contextmanager contract was implemented using the contextlib.contextmanager decorator; now it is explicitly implemented using __enter__ and __exit__.

File: tests/common/mock_cgroup_environment.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
```
Comment (narrieta) [inline tests/common/mock_cgroup_environment.py:1]: This file was renamed from mock_cgroup_environment.py. The _MOCKED* arrays stay the same in the new file and the mocking logic is now moved to mock_environment.py

**Review by larohra — APPROVED** [review]
Awesome thanks for adding this functionality, I'm sure it'll be super helpful in the future!

**Pull request merged**